pyi.py

```diff
@@ -270,6 +270,27 @@ def _is_bad_TypedDict(node: ast.Call) -> bool:
     )
 
 
+@dataclass
+class AttributeManager:
+    """Context manager to help the PyiVisitor keep track of internal state"""
+
+    nesting: int = 0
+
+    def __enter__(self) -> None:
+        self.nesting += 1
+
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_inst: BaseException | None,
+        exc_tb: TracebackType | None,
+    ) -> None:
+        self.nesting -= 1
+
+    def __bool__(self) -> bool:
+        return bool(self.nesting)
```

Comment:
I think an operator overload is a bit too clever for this. How about `is_active()` or something?